daemon/cluster/executor/container/controller.go

```diff
@@ -317,8 +317,10 @@ func (r *controller) Shutdown(ctx context.Context) error {
 
 	// remove container from service binding
 	if err := r.adapter.deactivateServiceBinding(); err != nil {
-		log.G(ctx).WithError(err).Errorf("failed to deactivate service binding for container %s", r.adapter.container.name())
-		return err
+		log.G(ctx).WithError(err).Warningf("failed to deactivate service binding for container %s", r.adapter.container.name())
+		// Don't return an error here, because failure to deactivate
+		// the service binding is expected if the container was never
+		// started.
```

Comment:
I agree we should not block the shutdown because a failure in the deactivate service binding, unless user is given a command to fix it. I was just looking at it, while I came to know from @dongluochen this was taken care in this PR as a side change.

Given this PR may take some time to go through, I am also thinking it would be good to have a separate PR which can be quickly merged.